test/regression_test.jl

```diff
@@ -47,52 +82,35 @@ R = MitosisStochasticDiffEq.Regression(sdekernel,yprototype,ϕprototype,paramjac
 
 
 Π = []
+Π2 = []
+
+G = MitosisStochasticDiffEq.conjugate(R, sol, 0.1*I(2))
+G2 = conjugate_posterior(sol, 0.1*I(2))
+
+mu = G.F
+Gamma = G.Γ
+WL = (cholesky(Hermitian(Gamma)).U)'
+
+Random.seed!(1)
 for i=1:K
-  G = MitosisStochasticDiffEq.conjugate(R, sol, 0.1*I)
-  mu = G.F
-  Gamma = G.Γ
-  WL = (cholesky(Hermitian(Gamma)).U)'
   th° = WL'\(randn(size(mu))+WL\mu)
   push!(Π,th°)
+  th° = WL'\(randn(size(mu))+WL\mu)
+  push!(Π2,th°)
 end
 
+mu = G2.F
+Gamma = G2.Γ
+WL = (cholesky(Hermitian(Gamma)).U)'
 
-
-"""
-    conjugate_posterior(Y, Ξ)
-
-Sample the posterior distribution of the conjugate drift parameters from path `Y`,
-prior precision matrix `Ξ` with non-conjugate parameters fixed in the model.
-Adjusted from http://www.math.chalmers.se/~smoritz/journal/2018/01/19/parameter-inference-for-a-simple-sir-model/
-for Bridge.jl
-"""
-function conjugate_posterior(Y, Ξ)
-    paramgrad(t, u) = [u, 1]
-    paramintercept(t, u) = 0
-    t, y = Y.t[1], Y.u[1]
-    ϕ = paramgrad(t, y)
-    mu = zero(ϕ)
-    G = zero(mu*mu')
-
-    for i in 1:length(Y)-1
-        ϕ = paramgrad(t, y)'
-        Gϕ = pinv(Y.prob.g(y, Y.prob.p, t)*Y.prob.g(y, Y.prob.p, t)')*ϕ # a is sigma*sigma'. Todo: smoothing like this is very slow
-        zi = ϕ'*Gϕ
-        t2, y2 = Y.t[i + 1], Y.u[i + 1]
-        dy = y2 - y
-        ds = t2 - t
-        #@show size(mu), size(Gϕ'), (dy - paramintercept(t, y)*ds)
-        mu = mu + Gϕ'*(dy - paramintercept(t, y)*ds)
-        t, y = t2, y2
-        G = G +  zi*ds
-    end
-    WW = G + Ξ
-    WL = (cholesky(Hermitian(WW)).U)'
-    th° = WL'\(randn(size(mu))+WL\mu)
+Random.seed!(1)
+for i=1:K
+  th° = WL'\(randn(size(mu))+WL\mu)
+  push!(Π2,th°)
 end
 
 
-Π2 = [conjugate_posterior(sol, 0.1*I) for i in 1:K]
+
 
 @test par[1:2] ≈ mean(Π) rtol=0.2
 @test par[1:2] ≈ mean(Π2) rtol=0.2
```

Comment:
can we actually add some more tests here .. like some restrictions on `cov(Π)`?

Reply:
The plan is to add a test comparing with https://github.com/mschauer/Mitosis.jl/blob/master/src/regression.jl